### ISSUE TITLE
Major enhancements to multivariate MODE

### ISSUE THREAD

**Issue by davidalbo** [dtcenter/MET#2745]
## Describe the Enhancement ##
Several items did not make it into met12.0 beta2 that can be done for beta3:

### Time Estimate ###
3-5 days

### Sub-Issues ###
- [x] Add the original mvmode snow test case to the unit tests. Several configurations should be added.
- [x] Look further into the percentile thresholding to correctly implement more of the options and exclude those that cannot be done in mvmode.
- [x] Document the percentile threshold options that require both a forecast and an obs input, indicating that the indices must match in those cases.
- [x] Fix file_type so that it can be anywhere in the config, from innermost individual input to all the way out.
- [x] Consider multiple convolution thresholds and radii as an option for mvmode (currently disallowed).
- [x] Consider quilting as an option for mvmode (currently disallowed).
- [x] Add more checking for inconsistent/incorrect mvmode config settings. 
- [ ] Discuss mode with A.I., how would that be implemented (not anticipating actually doing anything besides discussing)

### Relevant Deadlines ###
beta3

### Funding Source ###
Split (for Dave):
2773543 NOAA BIL METp SW   40
2799991 Met Office METplus Support  40

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [x] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Review default **alert** labels
- [x] Select **component(s)**
- [x] Select **priority**
- [ ] Select **requestor(s)**

### Milestone and Projects ###
- [x] Select **Milestone** as the next official version or **Backlog of Development Ideas**
- [x] For the next official version, select the **MET-X.Y.Z Development** project

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [x] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [x] Complete the development and test your changes.
- [x] Add/update log messages for easier debugging.
- [x] Add/update unit tests.
- [x] Add/update documentation.
- [x] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the next official version
Select: **MET-X.Y.Z Development** project for development toward the next official release
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

**Comment by davidalbo** [comment]
The unit tests have been added and do work on seneca.  Will need to do something extra to get things working when unit tests run as part of say a pull request. 

Percentile thresholding now has been implemented when appropriate, which includes every possibility except climatology options.  For the FBIAS option, the implementation uses the matching index in the other field, for example if fcst input 2 has  FBIAS0.9,  it uses obs input 2. This might require reordering of the inputs. 

Similarly SFP and SOP options for percentile thresholding can require both a forecast and an obs input, again the implementation uses the matching index in the other field if indicated.

Documenting FBIAS, SFP and SOP for multivariate mode is still needed.

file_type has been changed so that each input can have its own file_type.

Efficiency has been improved.  Each input is read from disk only once now, whereas before multiple reads happened.

I've decided against multiple threshold/radii and quilting.  Things are plenty complex as it is.
